code/modules/mob/living/carbon/xenomorph/castes/warrior/abilities_warrior.dm

```diff
@@ -177,7 +184,7 @@
 		return FALSE
 
 /datum/action/xeno_action/activable/fling/use_ability(atom/A)
-	var/mob/living/carbon/xenomorph/X = owner
+	var/mob/living/carbon/xenomorph/warrior/X = owner
```

Comment:
abilities should be portable on other xenos without too much trouble, so no cast